File: src/main/resources/static/scripts/customize/service/commentService.js
app.service('commentService', function ($http) {
    this.paginationComment = function (page, pageSize) {
        return $http.get("/comment/pagination?page=" + page + "&pageSize=" + pageSize);
    };
    this.newComment = function (entity) {
        return $http.post("/comment/newComment", entity);
    };
    this.deleteComment = function (id) {
        return $http({
            method: "DELETE",
            url: "/comment/deleteComment?id=" + id,
            data: {},
            headers: {"Content-Type": "application/json"}
        })
    };
    this.listByArticleId = function (id) {
        return $http.get("/comment/listByArticleId?id=" + id);
    }
});

File: src/main/java/per/neal/blog/service/CommentService.java
package per.neal.blog.service;

import per.neal.blog.entity.TbComment;
import per.neal.blog.entity.dto.PageResult;

import java.util.List;

/**
 * 评论service接口
 *
 * @author neal
 */
public interface CommentService {
    /**
     * 统计该ID下的评论条数
     *
     * @param id 文章ID
     * @return count
     */
    int countByArticleId(long id);

    /**
     * 根据文章id统计文章下的评论
     *
     * @param id 文章ID
     * @return list
     */
    List<TbComment> listByArticleId(long id);

    /**
     * 根据地址搜索分页查询
     *
     * @param page     当前页
     * @param pageSize 页面大小
     * @return PageResult
     */
    PageResult paginationComment(int page, int pageSize);

    /**
     * 保存评论
     *
     * @param comment TbComment
     */
    void saveComment(TbComment comment);

    /**
     * 删除评论
     *
     * @param id 评论ID
     */
    void deleteComment(long id);
}
